storagemarket/provider.go

```diff
@@ -94,19 +88,21 @@ type Provider struct {
 	Transport      transport.Transport
 	fundManager    *fundmanager.FundManager
 	storageManager *storagemanager.StorageManager
-	dealPublisher  *DealPublisher
+	dealPublisher  types.DealPublisher
 	transfers      *dealTransfers
 
-	secb                        *sectorblocks.SectorBlocks
+	pieceAdder                  types.PieceAdder
 	maxDealCollateralMultiplier uint64
 
 	fullnodeApi v1api.FullNode
 
-	dhsMu sync.RWMutex
-	dhs   map[uuid.UUID]*dealHandler
+	dhsMu  sync.RWMutex
+	dhs    map[uuid.UUID]*dealHandler
+	helper types.MinerHelper
```

Comment:
nit: helper isn't protected by dhsMu so let's move it out from under the ["mutex hat"](https://dmitri.shuralyov.com/idiomatic-go#mutex-hat)